### DISCUSSION TITLE
[Feature request] Installation script for Debian x64 platform

### DISCUSSION THREAD

**Discussion by nxtester** [raspiblitz/raspiblitz#3925] [Feature / App Ideas]
To push adoption and amount of Lightning full nodes running with RaspiBlitz we need beside the Raspberry Pi as hardware platform a more universal and tested, official installation script of RaspiBlitz for a recent Debian x64 platform. For example, Debian is widely adopted in virtual server hosting. personal computer x64 platforms and NAS with virtualization apps like QNAP and Synology.

I myself tried also several platforms and feel quite a bit frustrated. Whilst the Raspberry-Pi platform might be ideal for the RaspiBlitz installation, it seems that hardware isn't. Serious bottleneck is limited robustness of any MicroSD memorycard keeping OS and RaspiBlitz software. It's simply not designed for such usecase and therefore stability problems are daily routine. On the other hand, RaspiBlitz installation script always failed with Debian 11 x64 because of several reasons.

RaspiBlitz running with Raspberry-Pi is great for first steps but obviously not a robust enough 24/7/365 solution.

That's why a approved and official Debian x64 installation script for RaspiBlitz would be great.

**Comment by openoms** [comment]
See the work done towards running in an Ubuntu 20.04 VM: https://github.com/rootzoll/raspiblitz/issues/1500#issuecomment-917444439

Debian 11 is unfortunately not running on my TrueNAS based setup.

**Comment by openoms** [comment]
There is progress on the build script:
https://github.com/rootzoll/raspiblitz/issues/1500#issuecomment-982606560

The logic is that the scripts would use the second (nonOS) disk for the chain and other storage.

I am running VMs on Ubuntu, but testing on Debian is welcome.

**Comment by adevoss** [comment]
Testing on Debian VM #2772

**Comment by openoms** [comment]
Successfully installed raspiblitz using the build_script from the dev branch.
1 disk for the OS and used a separate 1TB one for the blockchain.

```
# download the build script
wget https://raw.githubusercontent.com/rootzoll/raspiblitz/dev/build_sdcard.sh
# run
sudo bash build_sdcard.sh false false rootzoll dev hdmi false
```

**Comment by openoms** [comment]
with the new changes in the build options it should be:
```
# download the build script
wget -O build_sdcard.sh https://raw.githubusercontent.com/rootzoll/raspiblitz/dev/build_sdcard.sh
# run
sudo bash build_sdcard.sh --github-user rootzoll --branch dev --display hdmi --tweak-boot-drive 0  --wifi-region off
```

**Comment by tioneb22** [comment]
Hey @openoms , I tried running the above from a Debian 11 VM but I get a "403 Forbidden" when trying to access the nginx page. I have a 32GB OS drive on /dev/sda1 and my blockchain drive (/dev/sdb1) is mounted to /mnt/hdd. I run the script and nothing gets populated in the /mnt/hdd folder. Not sure what I'm missing... any help is greatly appreciated.

**Comment by openoms** [comment]
connect with ssh to your VM

**Comment by tioneb22** [comment]
> connect with ssh to your VM

I was connected via SSH. I couldn't figure out how to get the scripts to generate data inside /mnt/hdd. I had mounted it manually and added an entry inside of /etc/fstab.

**Comment by openoms** [comment]
when you boot the OS image it simply wait for a second disk to be attached. Should work on the fly, but try restarting the VM if not picked up automatically.

**Comment by nxtester** [comment]
Folks, could anybody confirm a successful installation of RaspiBlitz in HyperV?  I think it's not possible for now.

**Comment by adevoss** [comment]
Managed to install v1.9.0 on Debian Bullseye using the `build_sdcard.sh` script. Steps:

- Detach datadrive
- Install OS on two partitions (/boot and /). Don't know if that's required but PiOS has these two.
- Download `build_sdcard.sh` version 1.9.0 and make executable
- Start script: `sudo ./build_sdcard.sh --interaction true --fatpack false --github-user rootzoll --branch v1.9.0 --display headless --tweak-boot-drive false --wifi-region off`

One workaround required:
Script `blitz.display.sh` does not support amd64 and stops execution with `exit 1` so I edited that script by commenting the lines `exit 1` below `baseimage not supported` lines.

- Execute `sudo blitz.fatpack.sh`

Continue with
- Shutdown
- Attach datadrive
- Boot
- Login with user admin

Migration starts

**Comment by adevoss** [comment]
:+1: for all devs of 1.9.0
